lib/dao.js

```diff
@@ -757,12 +757,12 @@ DataAccessObject.findOrCreate = function findOrCreate(query, data, options, cb)
 
         // options has precedence on model-setting
         if (options.validate === false) {
-          _findOrCreate(query, data, obj);
+          return _findOrCreate(query, data, obj);
```

Comment:
Oh, I see! In that case it's ok to not include any additional tests. I think we should strive to have all commits leaving the repo in a state that passes all unit-tests. In this case, it means reordering the two commits - first fix the issue, then add the new changes that would fail without the fix. You should be able to easily reorder commits via `git rebase -i`